src/lib/components/Swap/SwapButton.tsx

```diff
@@ -76,6 +77,12 @@ export default function SwapButton({ disabled }: SwapButtonProps) {
     })
   }, [addTransaction, getApproval])
 
+  const {
```

Comment:
Can you move this into useSwapInfo, and return a swap type enum saying if it's a swap, wrap, or unwrap? That way, the toolbar and swap button don't need to both call useWrapCallback - it's all taken care of in one place.

Reply:
I think doing so would also allow you to short-circuit a fetch if only a wrap is needed.